### ISSUE TITLE
Throws an error when parsing a do block with a wrapper function.

### ISSUE THREAD

**Issue by mjhm** [za-creature/coffeescope#24]
This throws an error

```
afn = (fn) -> fn()

do afn ->
  null
```

while the following doesn't

```
afn = (fn) -> fn()

afn( ->
  null
)()
```

The both compile identically.  The stack trace for the error looks like:

```
TypeError: Cannot read property 'constructor' of undefined
  at ScopeLinter.module.exports.ScopeLinter.visit (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:95:36)
  at ScopeLinter.visit (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at /Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:193:24
  at ScopeLinter.module.exports.ScopeLinter.newScope (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:79:9)
  at ScopeLinter.newScope (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at ScopeLinter.module.exports.ScopeLinter.newScope (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:84:16)
  at ScopeLinter.newScope (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at ScopeLinter.module.exports.ScopeLinter.lint (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:46:14)
  at ScopeLinter.lint (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/ScopeLinter.js:5:59)
  at Coffeescope2.module.exports.Coffeescope2.lintAST (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeescope2/lib/index.js:34:38)
  at ASTLinter.module.exports.ASTLinter.lint (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/coffeelint.js:613:43)
  at Object.coffeelint.lint (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/coffeelint.js:370:67)
  at ErrorReport.module.exports.ErrorReport.lint (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/coffeelint.js:758:51)
  at lintFiles (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/commandline.js:98:19)
  at Object.<anonymous> (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/commandline.js:251:21)
  at Object.<anonymous> (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/lib/commandline.js:256:4)
  at Module._compile (module.js:409:26)
  at Object.Module._extensions..js (module.js:416:10)
  at Module.load (module.js:343:32)
  at Function.Module._load (module.js:300:12)
  at Module.require (module.js:353:17)
  at require (internal/module.js:12:17)
  at Object.<anonymous> (/Users/johnmclaughlin/git/test_coffeelint/node_modules/coffeelint/bin/coffeelint:34:5)
  at Module._compile (module.js:409:26)
  at Object.Module._extensions..js (module.js:416:10)
  at Module.load (module.js:343:32)
  at Function.Module._load (module.js:300:12)
  at Function.Module.runMain (module.js:441:10)
  at startup (node.js:139:18)
  at node.js:968:3
```

**Comment by za-creature** [comment]
I can see that it crashes, but that code sample is scrambling my brain.

Could you elaborate please?

**Comment by za-creature** [comment]
Released in 0.3.10 under the assumption that

``` coffeescript
do fn
```

is equivalent (with respect to precedence) to

``` coffeescript
fn()
```

**Comment by mjhm** [comment]
@za-creature Thanks so much for the quick fix.  The real life use case for this is a construct that is starting to come up with promises and generators.  For example

```
do Promise.coroutine -> 
  aresult = yield aPromiseGeneratingFn()
  bresult = yield bPromiseGeneratingFn()
.catch (err) ->
  someErrorHandler err
```

`Promise.coroutine` wraps the generator function so that it returns a promise.
